lib/jison.js

```diff
@@ -1280,7 +1280,12 @@ function commonjsMain (args) {
         console.log('Usage: '+args[0]+' FILE');
         process.exit(1);
     }
-    var source = require('fs').readFileSync(require('path').normalize(args[1]), "utf8");
+    var source = null;
+    var fs = require('fs');
+    var path = require('path');
+    if (typeof fs !== 'undefined' && fs !== null && typeof path !== 'undefined' && path !== null) {
+        source = fs.readFileSync(path.normalize(args[1]), "utf8");
+    }
     return exports.parser.parse(source);
```

Comment:
And given that you want to code this defensively, wouldn't it be better to code it as a feature detect, i.e. along the lines of `if (fs && typeof fs.readFileSync === 'function' && path && typeof path.normalize === 'function') { source = ...`?

Besides, now that you start with an empty string a 'default input if fs/path don't exist in my env', which would very probably throw a parse error (on empty input) anyway, why the `source = ''` change? Right now you have a situation where, if readFilesync doesn't deliver, you end up with unusable input anyway (it'll always execute `parser.parse("")` then) so I wonder what you want to accomplish in a non-`readFileSync` environment. Maybe 'overload' the interface where 

```
if (typeof args === 'string') {
  // not using readFileSync, but feeding it the input straight away:
  return exports.parser.parse(args);
} else {
  if (readFileSync does not exist || args[] is not array) {
    throw Error('barfing hairball, mentioning unexpected environment situation?');
  } else {
    return ...parse(source); // loaded from file: args[1], i.e. args MUST be array
  }
}
```

That would at least make the CommonJS interface usable in both situations and add meaning to your checks; right now, all I see is a change in the *way* this will crash/fail.  😕